### PR TITLE
feat(object): add support for bucket lifecycle configuration

### LINKED ISSUE #4160: Add support for managing lifecycle of objects from the cli
```
aws s3api put-bucket-lifecycle-configuration --bucket my-bucket --lifecycle-configuration file://incomplete-upload-lifecycle-policy.json
```

put/get/delete

### PR DESCRIPTION
**Pull request by pypaut**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request.
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #4160 

Release note for [CHANGELOG](https://github.com/scaleway/scaleway-cli/blob/master/CHANGELOG.md):

```release-note
object: add support for buckets lifecycle configuration management
```

### CODE REVIEW

**Comment by estellesoulard** [comment]
(side-note regarding the description: that [CHANGELOG](https://github.com/scaleway/scaleway-cli/blob/master/CHANGELOG.md) file is properly obsolete (and probably misleading), we might want to remove it altogether in another PR :see_no_evil: )

File: internal/namespaces/object/v1/custom_bucket_lifecycle_configuration.go [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+	ProjectID              string
+}
+
+func lifecycleConfigurationCreateCommand() *core.Command {
+	return &core.Command{
+		Namespace: "object",
+		Resource:  "bucket-lifecycle",
+		Verb:      "create",
+		Short:     "Create a lifecycle configuration for an S3 bucket's objects.",
+		Long:      "Create a lifecycle configuration and apply to an Object Bucket's objectswith the S3 protocol.",
```
Comment (estellesoulard) [inline internal/namespaces/object/v1/custom_bucket_lifecycle_configuration.go]:
```suggestion
		Long:      "Create a lifecycle configuration and apply to an Object Bucket's objects with the S3 protocol.",
```

File: cmd/scw/testdata/test-all-usage-object-bucket-lifecycle-configuration-create-usage.golden [hunk 2]
```diff
@@ -0,0 +1,22 @@
+🎲🎲🎲 EXIT CODE: 0 🎲🎲🎲
+🟩🟩🟩 STDOUT️ 🟩🟩🟩️
+Create a lifecycle configuration and apply to an Object Bucket's objectswith the S3 protocol.
+
+USAGE:
+  scw object bucket-lifecycle-configuration create <bucket ...> [arg=value ...]
```
Comment (estellesoulard) [inline cmd/scw/testdata/test-all-usage-object-bucket-lifecycle-configuration-create-usage.golden]: Is this from a previous version? Shouldn't it be `bucket-lifecycle` ?
  
`./scw  object bucket-lifecycle-configuration
Object-storage utils

USAGE:
  ./scw object <command>

AVAILABLE COMMANDS:
  bucket           Manage S3 buckets
  bucket-lifecycle Manage S3 buckets' lifecycle configuration
  config           Manage configuration files for popular S3 tools
`

Comment (pypaut) [reply cmd/scw/testdata/test-all-usage-object-bucket-lifecycle-configuration-create-usage.golden]: Yes, remnants of the latest rename, deleting!

File: internal/namespaces/object/v1/custom_bucket_lifecycle.go [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+		Verb:      "create",
+		Short:     "Create a lifecycle configuration for an S3 bucket's objects.",
+		Long:      "Create a lifecycle configuration and apply to an Object Bucket's objects with the S3 protocol.",
+		ArgsType:  reflect.TypeOf(lifecycleCreateArgs{}),
+		ArgSpecs: core.ArgSpecs{
+			{
+				Name:       "bucket",
+				Positional: true,
+				Required:   true,
+				Short:      "The name of the bucket to which assign the policy.",
```
Comment (estellesoulard) [inline internal/namespaces/object/v1/custom_bucket_lifecycle.go]:
```suggestion
				Short:      "The name of the bucket to which assign the lifecycle configuration.",
```

File: internal/namespaces/object/v1/custom_bucket_lifecycle_test.go [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+      "Filter": {},
+      "AbortIncompleteMultipartUpload": {
+        "DaysAfterInitiation": 7
+      }
+    }
+  ]
+}`
+)
+
+func Test_BucketLifecycleCreate(t *testing.T) {
```
Comment (estellesoulard) [inline internal/namespaces/object/v1/custom_bucket_lifecycle_test.go:49]: Idk if it's pertinent (feel free to close if not), but should we add a test with empty `{"Rules": []}` ? Should we have validation on config content when unmarshalling in `lifecycleCreateCommand()` l.71?

Comment (pypaut) [reply internal/namespaces/object/v1/custom_bucket_lifecycle_test.go:49]: The CLI unmarshal will show fields errors, but most of the actual validation is done on the backend side. Added tests with empty configurations 👌🏻

**Review by estellesoulard — DISMISSED** [review]

**Review by estellesoulard — APPROVED** [review]
